UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs

```diff
@@ -45,14 +51,28 @@ public class ProductDataService : IProductDataService
             this.mapper = mapper;
             this.fileShareService = fileShareService;
             this.logger = logger;
-            this.exchangeSetStorageProvider = exchangeSetStorageProvider;          
+            this.exchangeSetStorageProvider = exchangeSetStorageProvider;
+            this.azureAdB2CConfiguration = azureAdB2CConfiguration;
+            this.azureAdConfiguration = azureAdConfiguration;
+            this.essFulfilmentStorageconfig = essFulfilmentStorageconfig;
         }
 
-        public async Task<ExchangeSetServiceResponse> CreateProductDataByProductIdentifiers(ProductIdentifierRequest productIdentifierRequest)
+        public async Task<ExchangeSetServiceResponse> CreateProductDataByProductIdentifiers(ProductIdentifierRequest productIdentifierRequest, AzureAdB2C azureAdB2C)
         {
             var salesCatalogueResponse = await salesCatalogueService.PostProductIdentifiersAsync(productIdentifierRequest.ProductIdentifier.ToList());
             ////can check for file size from salesCatalogueResponse.ResponseCode
```

Comment:
Can we add a check if (salesCatalogueResponse.ResponseCode == HttpStatusCode.OK) then only we will check azureb2c user and filesize